docs/code/IDataViewTypeSystem.md

```diff
@@ -661,154 +605,99 @@ This chapter contains information on the C# classes used to represent column
-singleton-class for the standard datetime timezone type. The instance is
-exposed by the static `DateTimeType.Instance` property.
+### `TimeSpanDataViewType` Sealed Class
 
-### `TimeSpanType` Sealed Class
+The `TimeSpanDataViewType` sealed class derives from `PrimitiveDataViewType`
+and is a singleton-class for the standard datetime timezone type. The instance
+is exposed by the static `TimeSpanDataViewType.Instance` property.
 
-The `TimeSpanType` sealed class derives from `PrimitiveType` and is a
-singleton-class for the standard datetime timezone type. The instance is
-exposed by the static `TimeSpanType.Instance` property.
+### `KeyDataViewType` Sealed Class
 
-### `KeyType` Sealed Class
-
-The `KeyType` sealed class derives from `PrimitiveType` and instances
+The `KeyDataViewType` sealed class derives from `PrimitiveType` and instances
 represent key types.
 
-Notes:
-
-* Two key types are considered equal iff their kind, min, count, and
-  contiguous values are the same.
-
-* The static `IsValidDataKind` method returns true iff kind is `U1`, `U2`,
-  `U4`, or `U8`. These are the only valid underlying data kinds for key types.
-
-* The inherited `KeyCount` property returns the same value as the `Count`
-  property.
+Note that two key types are considered equal iff their kind and count are the
+same, though they may be different instances. (Unlike many of the
+aforementioned types, which are based on singleton instances.)
 
 ### `VectorType` Sealed Class
 
-The `VectorType` sealed class derives from `ColumnType` and instances
+The `VectorType` sealed class derives from `DataViewType` and instances
 represent vector types. The item type is specified as the first parameter to
 each constructor and the dimension information is inferred from the additional
 parameters.
 
-* The `DimCount` property indicates the number of dimensions and the `GetDim`
-  method returns a particular dimension value. All dimension values are non-
-  negative integers. A dimension value of zero indicates unknown (or variable)
-  in that dimension.
+* Thd `Dimensions` property is an immutable array of `int` indicating the
+  dimensions of the vector. This will be of length at least one. All dimension
+  values are non- negative integers. A dimension value of zero indicates
+  unknown (or variable) in that dimension.
 
 * The `VectorSize` property returns the product of the dimensions.
 
-* The `IsSubtypeOf(VectorType other)` method returns true if this is a subtype
-  of `other`, in the sense that they have the same item type, and either have
-  the same `VectorSize` or `other.VectorSize` is zero.
-
 * The inherited `Equals` method returns true if the two types have the same
   item type and the same dimension information.
 
-* The inherited `SameSizeAndItemType(ColumnType other)` method returns true if
-  `other` is a vector type with the same item type and the same `VectorSize`
-  value.
+* The inherited `SameSizeAndItemType(DataViewType other)` method returns true
```

Comment:
> The inherited `SameSizeAndItemType(DataViewType other)` method

SameSizeAndItemType is now internal. Should we remove it here?

Reply:
Yes, good catch -- I removed it from the type information catalog, but not this more discursive section. Let's change this.